python/legate_dataframe/experimental_ibis/schema.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Tuple
+
+import ibis.expr.datatypes as dt
+import ibis.expr.schema as sch
+from pylibcudf.types import DataType, TypeId
+
+from legate_dataframe import LogicalTable
+
+_to_plc_type_id = {
+    dt.Boolean: TypeId.BOOL8,
+    # dt.Null: ,
+    dt.String: TypeId.STRING,
+    # dt.Binary: ,
+    # dt.Date: type_id.TIMESTAMP_DAYS,  # not sure about these times/dates
+    # dt.Time: TypeId.TIMESTAMP_NANOSECONDS,
+    dt.Int8: TypeId.INT8,
+    dt.Int16: TypeId.INT16,
+    dt.Int32: TypeId.INT32,
+    dt.Int64: TypeId.INT64,
+    dt.UInt8: TypeId.UINT8,
+    dt.UInt16: TypeId.UINT16,
+    dt.UInt32: TypeId.UINT32,
+    dt.UInt64: TypeId.UINT64,
+    dt.Float32: TypeId.FLOAT32,
+    dt.Float64: TypeId.FLOAT64,
+}
+
+
+_to_plc_type = {i_t: DataType(plc_tn) for i_t, plc_tn in _to_plc_type_id.items()}
+
+_from_legate_df_type = {v: k for k, v in _to_plc_type.items()}
+
+
+def to_plc_type(ibis_type):
+    # TODO: Should we take nullability into account? (and if, how?)
```

Comment:
If `legate` supports non-nullable columns, then maybe, because there are performance benefits, but if not, I would either ignore or error if someone asks for a non-nullable column / dtype

Reply:
Ah, logically, all columns are nullable.  Sometimes we may know that there are nulls as an optimization.
So, I think that means that we can reject it here (it won't arise without user input as an optimization hint).  At least for now.